src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 4;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    private double getRandomDouble() {
+        return random.nextDouble() * 10;
+    }
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        switch (figureNumber) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(),getRandomDouble());
```

Comment:
```suggestion
                return getRandomCircle();
```
lets create private methods to get each figure and invoke it here